crates/bevy_hierarchy/src/child_builder.rs

```diff
@@ -110,6 +112,8 @@ fn update_old_parents(world: &mut World, parent: Entity, children: &[Entity]) {
     push_events(world, events);
 }
 
+/// Removes `children` from `parent`'s [`Children`], removing the component if ends up empty and
+/// removes [`Parent`] components from `children`.
```

Comment:
I think the following might improve comprehension, to me `children` here is not explicit enough.

Additionally, splitting in two sentences helps clarity.

```suggestion
/// Removes entities in `children` from `parent`'s [`Children`], removing the component if ends up empty.
/// Also removes [`Parent`] components from entities in `children`.
```